api/src/main/java/org/apache/iceberg/view/ViewVersion.java

```diff
@@ -65,8 +66,14 @@ public interface ViewVersion {
    *
    * @return the string operation which produced the view version
    */
-  @Value.Derived
+  @Value.Lazy
```

Comment:
`@Value.Derived` is eagerly computing the value before `check()` is executed, so we're switching to `@Value.Lazy`

Reply:
@nastra, why does this matter? All that would happen is that this is set to `null` and then the check method throws an exception.